src/routes/check-faculties.js

```diff
@@ -0,0 +1,49 @@
+const facultiesIds = require("../../assets/faculties.json");
+const faculties = require("../../assets/faculties_acronyms.json");
+
+const got = require("got");
+const express = require("express");
+const router = express.Router();
+const jsdom = require("jsdom");
+const { JSDOM } = jsdom;
+
+router.get("/", function(req, res) {
+    res.send(facultiesIds);
+});
+
+router.get("/check", function(req, res) {
+    try {
+        faculties.forEach((faculty) => {
+            const url = `https://sigarra.up.pt/${faculty}/pt/UCURR_GERAL.PESQUISA_UCS`;
+            got(url)
+                .then((response) => {
+                    const dom = new JSDOM(response.body);
+                    const scriptHtml = dom.window.document.querySelector(
+                        'script[data-lov-template-zone="search"]'
+                    ).innerHTML;
+                    const dom2 = new JSDOM(scriptHtml.toString());
+                    const facultyId = dom2.window.document.querySelector(
+                        '.form input[name="pv_search_inst_id"]'
+                    ).value;
+
+                    // outdated id
+                    if (facultyId !== facultiesIds[faculty]) {
+                        throw new Error(
+                            `Facultie id ´${faculty}' outdated in the backend storage`
+                        );
+                    }
+                })
+                .catch((error) => {
+                    console.error(`URL: ${url}:${error}`);
+                    return res.status(404).json({ success: false, error: error.message });
```

Comment:
I don't think this should be a 404 error, maybe a 500 server error?

Reply:
The failure is related to the sigarra webpage so I thought the only reason to fail would be when the page is not available?